src/PhpUnit/UriInterfaceTestsTrait.php

```diff
@@ -0,0 +1,290 @@
+<?php
+
+namespace Maks3w\Psr7Assertions\PhpUnit;
+
+use PHPUnit_Framework_Assert as Assert;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Provide PHPUnit test methods for Psr\Http\Message\UriInterface constraints.
+ */
+trait UriInterfaceTestsTrait
+{
+    /**
+     * @return Psr\Http\Message\UriInterface
+     */
+    abstract public function createUriObject($url);
```

Comment:
Second as you see here https://github.com/nyamsprod/Psr7Assertions/blob/features-uriinterface-test/src/PhpUnit/ServerRequestInterfaceTestsTrait.php#L81

Up to the comment test methods verify PSR7 compliant for the default object and after that start to use the public methods for change the status and verify the object returns the values expected